tensorflow_recommenders_addons/dynamic_embedding/python/ops/cuckoo_hashtable_ops.py

```diff
@@ -264,8 +281,43 @@ def insert(self, keys, values, name=None):
       values = ops.convert_to_tensor(values, self._value_dtype, name="values")
       with ops.colocate_with(self.resource_handle):
         # pylint: disable=protected-access
-        op = cuckoo_hashtable_ops.tfra_cuckoo_hash_table_insert(
-            self.resource_handle, keys, values)
+        op = cuckoo_ops.tfra_cuckoo_hash_table_insert(self.resource_handle,
+                                                      keys, values)
+    return op
+
+  def accum(self, keys, values_or_deltas, exists, name=None):
+    """Associates `keys` with `values`.
+
+      Args:
+        keys: Keys to accmulate. Can be a tensor of any shape.
+          Must match the table's key type.
+        values_or_deltas: values to be associated with keys. Must be a tensor of
+          the same shape as `keys` and match the table's value type.
+        exists: A bool type tensor indicates keys were exist or not.
```

Comment:
I think that the table already knows whether if the keys are existed . Could u plz tell why `exists` is required.

Reply:
`exists` here is only a  **snipshot** for the key status got in most recent looking up in this process,  when you call `accum`, the key status maybe changed by other worker process, for you know the hash table is multi-process safe.
If exists is same with the real status of key in table when executing `accum`, action will be taken, or ignored. If exists is `True`, the `values_or_deltas` will be treated as deltas, if `false`, the `values_or_deltas` will be treated as `value`